jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/AbstractRuntimeManager.java

```diff
@@ -125,6 +125,18 @@ public AbstractRuntimeManager(RuntimeEnvironment environment, String identifier)
         }
         this.executionErrorManager = new ExecutionErrorManagerImpl(storage);
         ((SimpleRuntimeEnvironment)environment).getEnvironmentTemplate().set(EnvironmentName.EXEC_ERROR_MANAGER, executionErrorManager);
+        if ((this instanceof SingletonRuntimeManager) && (environment instanceof SchedulerProvider)) {
```

Comment:
wouldn't it be better to place it in SingletonRuntimeManager? then we don't have to cast it any more